driver.py

```diff
@@ -594,6 +709,143 @@ def create_throughput_session(exp,
+    :type t_per_dataset: int
+    :return: Model reference to the aggregation session to launch
+    :rtype: Model
+    """
+    settings = exp.create_run_settings("./cpp-data-aggregation/build/aggregation_producer",
+                                       [str(_bytes), str(t_per_dataset)])
+    settings.set_tasks(nodes * tasks)
+    settings.set_tasks_per_node(tasks)
+    settings.update_env({
+        "SS_ITERATIONS": str(iterations)
+    })
+    # TODO replace with settings.set("nodes", condition==exp.launcher=="slurm")
+    if exp._launcher == "slurm":
+        settings.set_nodes(nodes)
+
+    name = "-".join((
+        "aggregate-sess-prod",
+        "N"+str(nodes),
+        "T"+str(tasks),
+        "DBN"+str(db_nodes),
+        "ITER"+str(iterations),
+        "TB"+str(_bytes),
+        "TPD"+str(t_per_dataset),
+        _get_uuid()
+        ))
+
+    model = exp.create_model(name, settings)
+    exp.generate(model, overwrite=True)
+    write_run_config(model.path,
+                client_total=tasks*nodes,
+                client_per_node=tasks,
+                client_nodes=nodes,
+                database_nodes=db_nodes,
+                database_cpus=db_cpus,
+                iterations=iterations,
+                tensor_bytes=_bytes,
+                t_per_dataset=t_per_dataset)
+    return model
+
+def create_aggregation_cons_session(exp,
+                                    nodes,
+                                    tasks,
+                                    db_nodes,
+                                    db_cpus,
+                                    iterations,
+                                    _bytes,
+                                    t_per_dataset,
+                                    c_threads):
+    """Create a Model to run a consumer for the aggregation scaling session
+
+    :param exp: Experiment object for this test
+    :type exp: Experiment
+    :param nodes: number of nodes for the synthetic aggregation application
+    :type nodes: int
+    :param tasks: number of tasks per node for the aggregation application
+    :type tasks: int
+    :param db_nodes: number of database nodes
+    :type db_nodes: int
+    :param db_cpus: number of cpus used on each database node
+    :type db_cpus: int
+    :param iterations: number of append/retrieve loops by the application
+    :type iterations: int
+    :param _bytes: size in bytes of tensors to use for aggregation scaling
+    :type _bytes: int
+    :param t_per_dataset: the number of tensors per dataset
+    :type t_per_dataset: int
+    :param c_threads:  the number of client threads to use inside of SR client
+    :rtype c_threads: int
+    :type t_per_dataset: int
+    :rtype: Model
+    """
+    settings = exp.create_run_settings("./cpp-data-aggregation/build/aggregation_consumer",
+                                       [str(nodes*tasks)])
+    #settings.set_tasks(1)
+    settings.set_tasks_per_node(1)
+    settings.set_cpus_per_task(c_threads*2)
```

Comment:
The `*2` is for hyperthreads? If so, for systems that have HT disabled, this should be controlled by an optional parameter.